File: android/src/main/java/com/audiocontext/context/ContextState.kt
package com.audiocontext.context

enum class ContextState {
  RUNNING,
  CLOSED,
  ;

  companion object {
    fun fromString(state: String): ContextState =
      when (state.lowercase()) {
        "running" -> RUNNING
        "closed" -> CLOSED
        else -> throw IllegalArgumentException("Unknown context state: $state")
      }

    fun toString(state: ContextState): String =
      when (state) {
        RUNNING -> "running"
        CLOSED -> "closed"
      }
  }
}

File: ios/context/ContextState/ContextState.m
#import "ContextState.h"

@implementation ContextState

+ (ContextStateEnum)fromString:(NSString *)state {
    NSString *lowercaseState = [state lowercaseString];
    if ([lowercaseState isEqualToString:@"running"]) {
        return ContextStateRunning;
    } else if ([lowercaseState isEqualToString:@"closed"]) {
        return ContextStateClosed;
    } else {
        @throw [NSException exceptionWithName:NSInvalidArgumentException
                                       reason:[NSString stringWithFormat:@"Unknown context state: %@", state]
                                     userInfo:nil];
    }
}

+ (NSString *)toString:(ContextStateEnum)state {
    switch (state) {
        case ContextStateRunning:
            return @"running";
        case ContextStateClosed:
            return @"closed";
    }
}

@end
